web/src/main/java/nl/tehdreamteam/se42/web/Application.java

```diff
@@ -17,36 +18,49 @@
 
     private static final Logger logger = LogManager.getLogger(Application.class.getSimpleName());
 
-    /**
-     * Starts all services.
-     *
-     * @param args The command-line arguments.
-     */
-    public static void main(String[] args) {
-        ServiceContainer container = new ServiceContainer();
-        container.addService(new SoapWebService());
+    private final CommandHandler handler;
+    private final ServiceContainer container;
 
-        try {
-            container.startAll();
-        } catch (Exception e) {
-            logger.fatal("Failed to start all services.", e);
-        }
+    private Application() {
+        container = new ServiceContainer();
+        container.addService(new SoapWebService());
+        container.startAll();
 
-        CommandHandler handler = new CommandHandler();
+        handler = new CommandHandler();
         handler.register(new StartServicesCommand(container));
         handler.register(new StopServicesCommand(container));
         handler.register(new ExitApplicationCommand(container));
-
-        startReadingCommandLine(handler);
     }
 
-    private static void startReadingCommandLine(CommandHandler handler) {
+    private void startReadingCLI() {
         try (Scanner sc = new Scanner(System.in)) {
-            String line;
-            while ((line = sc.nextLine()) != null) {
-                handler.handle(line);
-            }
+            processScannerInput(sc);
+        }
+    }
+
+    private void processScannerInput(Scanner sc) {
+        String line;
+        while ((line = sc.nextLine()) != null) {
+            processReadLine(line);
         }
     }
 
+    private void processReadLine(String line) {
+        try {
+            handler.handle(line);
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1166)